test/decl/subscript/subscripting.swift

```diff
@@ -310,3 +310,12 @@ struct S4 {
     }
   }
 }
+
+// SR-2575
+struct SR2575 {
+  subscript() -> Int {
+    return 1
+  }
+}
+
+SR2575().subscript() // expected-error{{type 'SR2575' has no member 'subscript'}}
```

Comment:
We should probably make a note / bug to come back and improve this specific case later.